waspc/src/Wasp/Generator/ServerGenerator/ApiRoutesG.hs

```diff
@@ -0,0 +1,68 @@
+module Wasp.Generator.ServerGenerator.ApiRoutesG
+  ( genApis,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import qualified Data.Aeson as Aeson
+import Data.Char (toLower)
+import StrongPath (Dir, File', Path, Path', Posix, Rel, reldirP, relfile)
+import qualified StrongPath as SP
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec as AS
+import Wasp.AppSpec.Api (Api)
+import qualified Wasp.AppSpec.Api as Api
+import Wasp.Generator.Common (ServerRootDir, makeJsonWithEntityData)
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.Monad (Generator)
+import qualified Wasp.Generator.ServerGenerator.Common as C
+import Wasp.Generator.ServerGenerator.JsImport (getJsImportStmtAndIdentifier)
+
+genApis :: AppSpec -> Generator [FileDraft]
+genApis spec =
+  sequence
+    [ genApiRoutes spec,
+      genApiTypes spec
+    ]
+
+genApiRoutes :: AppSpec -> Generator FileDraft
+genApiRoutes spec =
+  return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
+  where
+    apis = map snd $ AS.getApis spec
+    tmplData = object ["apiRoutes" .= map getTmplData apis]
+    tmplFile = C.asTmplFile [relfile|src/routes/apis/index.js|]
+    dstFile = SP.castRel tmplFile :: Path' (Rel ServerRootDir) File'
+
+    getTmplData :: Api -> Aeson.Value
+    getTmplData api =
+      let (jsImportStmt, jsImportIdentifier) = getJsImportStmtAndIdentifier relPathFromApisDirToServerSrcDir (Api.fn api)
+       in object
+            [ "routeMethod" .= map toLower (show $ Api.method . Api.httpRoute $ api),
+              "routePath" .= (Api.path . Api.httpRoute $ api),
+              "importStatement" .= jsImportStmt,
+              "importIdentifier" .= jsImportIdentifier,
+              "entities" .= getApiEntitiesObject api
+            ]
+
+genApiTypes :: AppSpec -> Generator FileDraft
+genApiTypes spec =
+  return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
+  where
+    namedApis = AS.getApis spec
+    tmplData = object ["apiRoutes" .= map getTmplData namedApis]
+    tmplFile = C.asTmplFile [relfile|src/apis/types.ts|]
+    dstFile = SP.castRel tmplFile :: Path' (Rel ServerRootDir) File'
+
+    getTmplData :: (String, Api) -> Aeson.Value
+    getTmplData (name, api) =
+      object
+        [ "name" .= name,
+          "entities" .= getApiEntitiesObject api
+        ]
+
+getApiEntitiesObject :: Api -> [Aeson.Value]
```

Comment:
Is there a plan to use the explicit entities list used by the api for something?

Reply:
or by letting them import them via normal imports. Hm I thought we have a reason for this but I also can't remember!

Reply:
Yeah, I'm not sure to be honest if we get any big gains from making it explicit (beyond looking like operations, and leaving us open to using that knowledge in the future somehow if needed). I don't know that we have a good way to let them import the Prisma query proxy object (which is different than the Prisma entity type) right now. I'll tinker with it and see if just giving them all entities in the context feels right 👍🏻 thnx for the idea

Reply:
I plan to leave it explicit in the wasp file for now, since it looks/feels like Operations and we may have a future use.